src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3051,6 +3060,21 @@
       <Kind Name="SemicolonToken"/>
     </Field>
   </Node>
+  <Node Name="SingleLineNamespaceDeclarationSyntax" Base="BaseNamespaceDeclarationSyntax">
```

Comment:
By convention we call these "directive" I suggest `NamespaceDirectiveSyntax`

Reply:
i don't actually like that.  because now you'd have:

```
BaseNamespaceDeclarationSyntax
|--NamespaceDeclarationSyntax
|--NamespaceDirectiveSyntax
```

We pick 'directive' for things that are very different (i.e. using statement vs directive).  In this case, these are the practically the same.  It's really a specialization of a core concept.  So i think the name should be SomethingNamespaceDeclaration.  Now, if that's SingleLineNamespaceDeclaration, or FileScopedNamespaceDeclaration, or something else is open for me.  But the latter part i believe is named properly :)

Reply:
Maybe call the base BaseNamespaceSyntax.

This is a directive as it affects whatever comes after, it is not a declaration in and of itself, there are a few other nodes in the language already with this semantic to make a comparison.

Reply:
i can def discuss this with the team.  Having:

```
NamepsaceSyntax
|--NamespaceDeclarationSyntax
|--NamespaceDirectiveSyntax
```

Is potentially possible.  I still dont' personally like it though.  I still see these as declarations, even if there is no body.  In other words, a declaration is not defined by having a body, but by defining a symbol (or symbols).

Reply:
>but by defining a symbol

I agree, I think you can look at it either way, it is a declaration as it defines a symbol but a directive syntactically. I think that hierarchy is expressive enough though and we don't have to think of another term to explain it.